make GraphQL API endpoints support fetching models from depot/metadata server

akphi opened the pull request:
#### What type of PR is this?

new feature

#### What does this PR do / why is it needed ?

Currently, GraphQL API endpoints only allow specifying the SDLC project coordinate to fetch model data, we would like to allow specifying pure model context data as input for more flexibility

This will work when https://github.com/finos/legend-depot/pull/139 is merged

pierredebelen commented:
I don't think a graphQL client would be able to send the metadata. Getting it from the 'depot' in Prod would be good to add though.

akphi commented:
> I don't think a graphQL client would be able to send the metadata. Getting it from the 'depot' in Prod would be good to add though.

Sorry, that's what I implemented, I just named the PR wrong. Fixed now.
